packages/cli/src/ui/utils/updateCheck.test.ts

```diff
@@ -85,7 +93,9 @@ describe('checkForUpdates', () => {
       version: '1.1.0',
     });
     updateNotifier.mockReturnValue({
-      update: { current: '1.1.0', latest: '1.0.0' },
+      fetchInfo: vi
+        .fn()
+        .mockResolvedValue({ current: '1.1.0', latest: '1.0.0' }),
     });
```

Comment:
please add a test case that verifies the case you fixed is resolved. for example, write a test that would have failed previously due to the race condition you fixed.